teachertool/src/state/helpers.ts

```diff
@@ -43,10 +44,24 @@ export function verifyRubricIntegrity(rubric: Rubric): {
     return { valid: invalidCriteria.length === 0, validCriteria, invalidCriteria };
 }
 
-export function isProjectLoaded(state: AppState) {
+export function isProjectLoaded(state: AppState): boolean {
     return !!state.projectMetadata;
 }
 
+export function isRubricLoaded(state: AppState): boolean {
+    return !!(state.rubric.criteria.length || state.rubric.name);
+}
+
+export function getSafeProjectName(state: AppState): string | undefined {
+    if (state.projectMetadata) {
+        return state.projectMetadata.name ?? Strings.UntitledProject;
+    }
+}
+
+export function getSafeRubricName(state: AppState): string | undefined {
```

Comment:
How would the return type here be undefined?

Reply:
Future-proofing for when we have a "My Rubrics" carousel.

Reply:
I still don't understand. Even if a rubric was created and listed in the "My Rubrics" carousel, won't they always have either a user-created name or a name that we generated to make sure all rubrics have names?

Reply:
If it works like the editor then they won't always have a rubric loaded. There's also the first-time user experience to consider.

Reply:
Sure, but in the current design of this function, there are one of two things getting returned:
1. The name of the rubric that the user made
2. "Untitled Rubric" 
So there is no possibility that this would return undefined unless we changed how this function worked. Or is there something I'm misunderstanding?

Reply:
In the very near future there will be a case where it returns undefined. This is future proofing for that eminent scenario.